tools/tsp-client/src/git.ts

```diff
@@ -2,14 +2,17 @@ import { spawn } from "child_process";
 import { simpleGit } from "simple-git";
 
 export async function getRepoRoot(repoPath: string): Promise<string> {
-  return await simpleGit(repoPath).revparse(["--show-toplevel"]);
+  return simpleGit(repoPath).revparse(["--show-toplevel"]);
 }
 
 export async function cloneRepo(rootUrl: string, cloneDir: string, repo: string): Promise<void> {
-  await simpleGit(rootUrl).clone(repo, cloneDir, ["--no-checkout", "--filter=tree:0"], (err) => {
-    if (err) {
-      throw err;
-    }
+  return new Promise((resolve, reject) => {
+    simpleGit(rootUrl).clone(repo, cloneDir, ["--no-checkout", "--filter=tree:0"], (err) => {
+      if (err) {
+        reject(err);
+      }
+    });
+    resolve();
```

Comment:
Is this the correct way to do this?

Reply:
I would think it is inside the callback. So if you don't fall into if(err){} I guess that means clone was succesful, right? If so you can resolve there
```suggestion
  return new Promise((resolve, reject) => {
    simpleGit(rootUrl).clone(repo, cloneDir, ["--no-checkout", "--filter=tree:0"], (err) => {
      if (err) {
        reject(err);
      }
      resolve();
    });
```